pkg/ccl/streamingccl/streamclient/client.go

```diff
@@ -51,6 +51,12 @@ type Client interface {
 	// can be used to interact with this stream in the future.
 	Create(ctx context.Context, tenant roachpb.TenantName) (streampb.ReplicationProducerSpec, error)
 
+	// StartAndPlanEphemeralSpanConfigsStream creates a stream for the span configs
+	// that apply to the passed in tenant, and returns the subscriptions the
+	// client can subscribe to. No protected timestamp or job is persisted to the
+	// source cluster.
+	StartAndPlanEphemeralSpanConfigsStream(ctx context.Context, tenant roachpb.TenantName) (streampb.StreamID, Topology, error)
```

Comment:
[nit] We don't need to make a final decision now, but I think that the naming here is a bit odd now that we aren't actually making any sort of producer-side job. Namely, this function doesn't _start_ anything.  And, in other parts of the code and comments we seem to want this call this `Create...` or `Setup...`.  Personally, I would go with something like `PlanSpanConfigsStream`.